# 3_PANDAS_EURES_CH

Analyse der Daten mit Hilfe von Pandas

### Schritt 1: Datenfile laden

In [28]:
import pandas as pd

In [29]:
path = "Data_EURES_CH_17022019.csv"

with open(path) as f:
    dateiinhalt = f.read()

In [30]:
df = pd.read_csv(path)

### Schritt 2: Grösse und Struktur der Tabelle 

In [42]:
len(df)

4831

*Resultat:* Die Tabelle hat 4831 Einträge, was der Anzahl Suchresultaten entspricht

In [40]:
df.head(3)

id   creationDate  lastModificationDate  \
0  MDgyQ05KQyA5  1547103651000         1550419220000   
1  MDgyWFJLUSA5  1548745236000         1550419219000   
2  MDgyV0xYUyA5  1548662527000         1550417428000   

                                              title  \
0  Conseiller / Conseillère clientèle en assurances   
1                   CONSEILLER A LA CLIENTELE (H/F)   
2                  Téléconseiller / Téléconseillère   

                                         description  numberOfPosts  \
0  Prospection téléphonique <br>Suivi de la clien...              1   
1  Analyser la situation globale de vos clients a...              1   
2  Notre club s'oriente aujourd'hui vers un club ...              1   

      locationMap  euresFlag                         jobCategoriesCodes  \
0  {'CH': [None]}      False  ['http://data.europa.eu/esco/isco/C3321']   
1  {'CH': [None]}      False  ['http://data.europa.eu/esco/isco/C3321']   
2  {'CH': [None]}      False  ['http://data.europa.eu/esco/isco/C5244']   

                                            employer  score  
0  {'name': 'PHOENIX MANAGEMENT', 'website': None...    NaN  
1  {'name': 'MF CONSEIL SARL', 'website': None, '...    NaN  
2  {'name': 'FRANCE LOISIRS SUISSE', 'website': N...    NaN

In [38]:
df.tail(3)

id   creationDate  lastModificationDate  \
4828  MTAwMDAtMTE2NDk0NTI5OC1TIDE  1535144224976         1535144224976   
4829  MTAwMDAtMTE0ODA0MDA2NS1TIDE  1531547424942         1531547424942   
4830  MTAwMDAtMTE0Nzg5ODUwOC1TIDE  1531547424251         1531547424251   

                                                  title  \
4828  Gesundheits- und Krankenpfleger/in auf der Gyn...   
4829  Facharzt (m/w) Innere Medizin (Facharzt/-ärzti...   
4830  Chefarzt (m/w) Ophtalmologie (Facharzt/-ärztin...   

                                            description  numberOfPosts  \
4828  Stellenangebotsbeschreibung: <br>Die Personal ...              2   
4829  Stellenangebotsbeschreibung: <br>Die Klinik su...              1   
4830  Stellenangebotsbeschreibung: <br>Die Klinik su...              1   

         locationMap  euresFlag  \
4828  {'CH': [None]}      False   
4829  {'CH': [None]}      False   
4830  {'CH': [None]}      False   

                                     jobCategoriesCodes  \
4828          ['http://data.europa.eu/esco/isco/C3221']   
4829          ['http://data.europa.eu/esco/isco/C2212']   
4830  ['http://data.europa.eu/esco/isco/C2212', 'htt...   

                                               employer  score  
4828  {'name': 'Personal Spezial Arzt und Pflege Ver...    NaN  
4829  {'name': 'Connecting People Medical', 'website...    NaN  
4830  {'name': 'Connecting People Medical', 'website...    NaN

In [33]:
df.dtypes

id                       object
creationDate              int64
lastModificationDate      int64
title                    object
description              object
numberOfPosts             int64
locationMap              object
euresFlag                  bool
jobCategoriesCodes       object
employer                 object
score                   float64
dtype: object

In [34]:
df.describe()

creationDate  lastModificationDate  numberOfPosts  score
count  4.831000e+03          4.831000e+03    4831.000000    0.0
mean   1.546273e+12          1.548625e+12       1.587663    NaN
std    5.156601e+09          1.661317e+09       1.819530    NaN
min    1.531547e+12          1.531547e+12       1.000000    NaN
25%    1.545318e+12          1.548108e+12       1.000000    NaN
50%    1.548223e+12          1.549062e+12       1.000000    NaN
75%    1.549323e+12          1.549922e+12       1.000000    NaN
max    1.550357e+12          1.550419e+12      40.000000    NaN

In [37]:
df['numberOfPosts'].sum()

7670

**Resultat 2:** Zum Teil sind pro Anzeige mehrere Stellen ausgeschrieben

In [53]:
df[df['numberOfPosts'] == 2]

id   creationDate  lastModificationDate  \
26        MTAwMDAtMTE2OTUxODU3NC1TIDE  1550356759584         1550356759584   
27        MTAwMDAtMTE2OTUxODA3OS1TIDE  1550356753314         1550356753314   
60            MTE4NTgtMjY2ODk0Mi1TIDE  1550273101082         1550273101082   
75    MTIxNDctMjAxOTAyMTU4MDc1Ny1TIDE  1550273004162         1550273004162   
76    MTIxNDctMjAxOTAyMTU4MDc0OS1TIDE  1550273004116         1550273004116   
81        MTAwMDAtMTE2OTQ5NTM0Ny1TIDE  1550272949522         1550272949522   
101       MTAwMDAtMTE2OTQxMjIzMi1TIDE  1550099955454         1550271933294   
108       MTAwMDAtMTE2ODYzNDAzMC1TIDE  1548198670303         1550271629139   
189                      MDgySktMTSA5  1547555593000         1550217670000   
254       MTAwMDAtMTE2OTQ1MTc4Ni1TIDE  1550187038106         1550187038106   
255       MTAwMDAtMTE2OTQ1MTE0OS1TIDE  1550187030564         1550187030564   
324       MTAwMDAtMTE0OTQ5NzY1Ny1TIDE  1531547462170         1550184992390   
325       MTAwMDAtMTE0OTQ5NzYxMy1TIDE  1531547462167         1550184992385   
488       MTAwMDAtMTE2OTQxMTg2OC1TIDE  1550099954712         1550099954712   
637       MTAwMDAtMTE2OTM3MTU5NC1TIDE  1550013887844         1550013887844   
638       MTAwMDAtMTE2OTM3MTU4MC1TIDE  1550013887081         1550013887081   
639       MTAwMDAtMTE2OTM3MTU0Ni1TIDE  1550013880745         1550013880745   
654       MTAwMDAtMTE2OTMxMDE5NC1TIDE  1549668567276         1550013115246   
688       MTAwMDAtMTE2ODk1MTE3MS1TIDE  1548716646481         1550012930518   
689       MTAwMDAtMTE2ODk0MTQzMy1TIDE  1548716579050         1550012921860   
691       MTAwMDAtMTE2ODcxOTIyNy1TIDE  1548112202494         1550012849080   
862       MTAwMDAtMTE2OTM2MjIzNC1TIDE  1549927957061         1549927957061   
863       MTAwMDAtMTE2OTM2MjE3MS1TIDE  1549927956968         1549927956968   
864       MTAwMDAtMTE2OTM2MTYzNy1TIDE  1549927955807         1549927955807   
865       MTAwMDAtMTE2OTM2MTQzOS1TIDE  1549927955312         1549927955312   
867       MTAwMDAtMTE2OTM2MTE5My1TIDE  1549927953883         1549927953883   
868       MTAwMDAtMTE2OTM2MDcwMy1TIDE  1549927947210         1549927947210   
935       MTAwMDAtMTE2ODY5MzQxNC1TIDE  1548111947133         1549926580063   
937       MTAwMDAtMTE2ODU5ODE4Ni1TIDE  1547680093055         1549926536900   
941       MTAwMDAtMTE2Nzg3OTIxMS1TIDE  1545088207085         1549926363147   
...                               ...            ...                   ...   
2743      MTAwMDAtMTE2ODgyNzAyMS1TIDE  1548371093518         1548889969580   
2753      MTAwMDAtMTE2MTkxOTM2MS1TIDE  1531550693982         1548889290727   
2829      MTAwMDAtMTE2ODk4NzgxNi1TIDE  1548803147147         1548803147147   
2857      MTAwMDAtMTE2ODA0Njg4OC1TIDE  1545433425555         1548802056665   
3003      MTAwMDAtMTE2ODQ5OTgyOC1TIDE  1547507563006         1548715606011   
3170      MTAwMDAtMTE2ODg4NzUyNy1TIDE  1548457720900         1548457720900   
3275      MTAwMDAtMTE2ODgzODQ5My1TIDE  1548371185485         1548371185485   
3283      MTAwMDAtMTE2NzU5NTI3NS1TIDE  1544052262589         1548369933626   
3313      MTAwMDAtMTE2ODc5NzQ4Ny1TIDE  1548285051339         1548285051339   
3381      MTAwMDAtMTE2ODcxNjY2Mi1TIDE  1548112176949         1548283697263   
3448      MTAwMDAtMTE2ODc2OTExNy1TIDE  1548198955666         1548198955666   
3449      MTAwMDAtMTE2ODc2OTA4OC1TIDE  1548198955586         1548198955586   
3451      MTAwMDAtMTE2ODc2ODk3NS1TIDE  1548198955325         1548198955325   
3558      MTAwMDAtMTE2ODcwNTg4OS1TIDE  1548112095748         1548112095748   
3839      MTAwMDAtMTE2ODU3ODkzMC1TIDE  1547679940814         1547679940814   
3841      MTAwMDAtMTE2NjY3NzgyNC1TIDE  1541369858138         1547678696919   
4380      MTAwMDAtMTE2ODE2MTI0My1TIDE  1546208566503         1546208566503   
4382      MTAwMDAtMTE2ODEzNjE0Mi1TIDE  1546037088169         1546037088169   
4387      MTAwMDAtMTE2ODEwNDgxOC1TIDE  1545953125999         1545953125999   
4409      MTAwMDAtMTE2ODA3MzE5OS1TIDE  1545634988369     

In [44]:
df.groupby('jobCategoriesCodes')['numberOfPosts'].mean().sort_values(ascending=False).head(3)

jobCategoriesCodes
['http://data.europa.eu/esco/isco/C9329']                                                                                      15.0
['http://data.europa.eu/esco/isco/C7124', 'http://data.europa.eu/esco/isco/C7121']                                             15.0
['http://data.europa.eu/esco/isco/C5151', 'http://data.europa.eu/esco/isco/C5131', 'http://data.europa.eu/esco/isco/C4224']    12.0
Name: numberOfPosts, dtype: float64

<font color='red'>**Weiteres Vorgehen**: Zum Teil sind unter einem Stelleninserat mehrere Jobs offen. Sind es die gleichen Arten von Jobs gibt es nur einen Link unter jobCategoriesCodes. Es kann aber auch sein dass mehrere Links aufgeführt werden. Die Links müssem in je einer Zeile dargestellt werden. Danach kann jobCategoriesCodes so gesplittet werden, dass es nur die erste Zahl nach dem C anzeigt, das ist die oberste Jobkategorie laut ISO-Norm. **Ziel** ist die Tabelle nach jobCategoriesCodes sortieren zu können.</font>

### Schritt 3: Daten reinigen

In [55]:
# Suchbegriff: Pandas split list in column/row into multiple rows


 df.join(s.apply(lambda x: Series(x.split(':'))))

# https://im-coder.com/pandas-wie-teile-ich-text-in-einer-spalte-in-mehrere-zeilen-auf.html

NameError: name 'Series' is not defined

### Schritt ???: Resultat abspeichern

<font color='red'>Gereingte Daten abspeichern. Damit damit geplottet werden kann **Jupyter Notebook: 3_PANDAS / PLOTEN**.</font>